dotCMS/src/integration-test/java/com/dotmarketing/portlets/htmlpages/business/render/HTMLPageAssetRenderedAPIImplIntegretionTest.java

```diff
@@ -0,0 +1,418 @@
+     * When Host is not set into request neither session
+     * it should take the default host
+     * Should return a right {@link PageView}
+     * @throws DotDataException
+     * @throws DotSecurityException
+     * @throws InterruptedException
+     */
+    @Test
+    public void shouldUseDefaultHostWhenHostIsNotSet()
+            throws DotDataException, DotSecurityException {
+        final Host defaultHost = APILocator.getHostAPI().findDefaultHost(APILocator.systemUser(), true);
+        init(defaultHost);
+        final PageMode pageMode = PageMode.ADMIN_MODE;
+        addPermission(role, defaultHost);
+
+        when(request.getAttribute(com.liferay.portal.util.WebKeys.USER)).thenReturn(user);
+
+        final HTMLPageAssetRenderedAPIImpl htmlPageAssetRenderedAPIImpl = new HTMLPageAssetRenderedAPIImpl();
+
+        final PageView pageRendered = htmlPageAssetRenderedAPIImpl.getPageRendered(
+                request, response, user, htmlPageAsset.getURI(), pageMode);
+
+        assertEquals(pageRendered.getPageInfo().getPage(), htmlPageAsset);
+        assertEquals(pageRendered.getSite(), defaultHost);
+    }
+
+
+    /**
+     * Method to test: {@link HTMLPageAssetRenderedAPIImpl#getPageRendered(PageContext, HttpServletRequest, HttpServletResponse)
+     * When Host is not set into request neither session
+     * it should take the request's server name before then the default host
+     * Should return a right {@link PageView}
+     * @throws DotDataException
+     * @throws DotSecurityException
+     * @throws InterruptedException
+     */
+    @Test
+    public void shouldUseRequestServerNameHostWhenHostIsNotSet()
+            throws DotDataException, DotSecurityException {
+        init(host);
+
+        when(request.getServerName()).thenReturn(host.getHostname());
+        final PageMode pageMode = PageMode.ADMIN_MODE;
+        addPermission(role, host);
+
+        when(request.getAttribute(com.liferay.portal.util.WebKeys.USER)).thenReturn(user);
+
+        final HTMLPageAssetRenderedAPIImpl htmlPageAssetRenderedAPIImpl = new HTMLPageAssetRenderedAPIImpl();
+
+        final PageView pageRendered = htmlPageAssetRenderedAPIImpl.getPageRendered(
+                request, response, user, htmlPageAsset.getURI(), pageMode);
+
+        assertEquals(pageRendered.getPageInfo().getPage(), htmlPageAsset);
+        assertEquals(pageRendered.getSite(), this.host);
+    }
+
+    /**
+     * Method to test: {@link HTMLPageAssetRenderedAPIImpl#getPageRendered(PageContext, HttpServletRequest, HttpServletResponse)
+     * When The host_id request's parameter is set, but the user does not have permission over it
+     * Should return a right {@link PageView}
+     * @throws DotDataException
+     * @throws DotSecurityException
+     * @throws InterruptedException
+     */
+    @Test(expected = DotSecurityException.class)
+    public void shouldUseHostIdParameterAndThrowDotSecurityException() throws DotDataException, DotSecurityException {
```

Comment:
All these methods that throw a DotSecurityException can be moved to a single test and use data providers